app/gui/src/controller/searcher.rs

```diff
@@ -739,22 +739,11 @@ impl Searcher {
                     prefix_id: default(),
                 };
                 expression.args.push(new_argument);
-                expression
+                Some(expression)
             }
         };
-        let expr_and_method = || {
-            let input_chain = Some(new_expression);
-
-            let expression = match (self.this_var(), input_chain) {
-                (Some(this_var), Some(input)) =>
-                    apply_this_argument(this_var, &input.wrapped.into_ast()).repr(),
-                (None, Some(input)) => input.wrapped.into_ast().repr(),
-                (_, None) => "".to_owned(),
-            };
-            let intended_method = self.intended_method();
-            (expression, intended_method)
-        };
-        let (expression, intended_method) = expr_and_method();
+        let expression = self.get_expression(new_expression);
```

Comment:
The name is unclear. Why, having `new_expression` we must yet get expression from it?

Additionally, I think `new_expression` construction is still duplicated.